### DISCUSSION TITLE
Save and reload the model.

### DISCUSSION THREAD

**Discussion by Xukaizh** [cornellius-gp/gpytorch#2565] [Q&A]
I have defined a GPModel use gpytorch, and saved it using:

import joblib
torch.save(model.state_dict(), 'gp_model.pth')

When I try to use the saved model, 

model = GPModel(train_x, train_y, likelihood, amplitude=5.0, frequency=8.0, phase=0.0).to(device)

If I set train_x and train_y into None, it gives different prediction result. But in a new file, there is no train_x and train_y defined. Do I need to redefine or are there any methods?

**Comment by Balandat** [comment]
If you have an exact GP, the `state_dict()` of the model only includes the model hyperparameters (e.g. lengthscales), but not the training data. You need to save the train input and targets separately, and then use `set_train_data()` to load that data into the model (in addition to calling `load_state_dict()`.
